server/src/main/java/org/opensearch/index/mapper/MappedFieldType.java

```diff
@@ -359,18 +359,31 @@ public Query phraseQuery(TokenStream stream, int slop, boolean enablePositionInc
         );
     }
 
+    public Query phraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements, QueryShardContext context) throws IOException {
+        return phraseQuery(stream, slop, enablePositionIncrements);
```

Comment:
This doesn't look right to me. Shouldn't the function on line 356 return this function. 

This function should really be 

```
public Query phraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements, QueryShardContext context) throws IOException {
throw new IllegalArgumentException(
            "Can only use phrase queries on text fields - not on [" + name + "] which is of type [" + typeName() + "]"
        );
    }
```

And the other overloaded function should be 
```
return phraseQuery(stream, slop, enablePositionIncrements, null);
```

Reply:
good callout, thanks.

Reply:
The reason I deviated from the standard method of overloading was to prevent any backward incompatibility changes to these public APIs, especially within the plugins. The current approach ensures that all mapper plugins overriding these methods will continue to function. In this pull request, my aim is to ensure that all callers of the method are directed to the new implementation, which, in turn, internally invokes the old implementation as the default behavior.